### ISSUE TITLE
Any plans for mircophone Noise supression?

### ISSUE THREAD

**Issue by joakim-tjernlund** [thesofproject/sof-bin#36]

**Comment by lgirdwood** [comment]
@joakim-tjernlund can you be more specific ? @singalsu fyi.

**Comment by joakim-tjernlund** [comment]
Not really, I only know that on our new Lenovo T14 with SOF one can hear some noise(white?) when using the built-in mic.

**Comment by plbossart** [comment]
@joakim-tjernlund please file an issue in the Linux tree and add your dmesg log with the following file added as /etc/modprobe.d/sof-dyndng.sof 
[sof-dyndbg.conf.txt](https://github.com/thesofproject/sof-bin/files/6206881/sof-dyndbg.conf.txt)

You may be experiencing known issues with the 2 mic support on some Lenovo platforms who use the PDM1 interface.

see e.g. https://github.com/thesofproject/linux/issues/2795

**Comment by singalsu** [comment]
@joakim-tjernlund There's a microphone beamformer and if properly set up it will attenuate a bit the noise but it also sets a fixed direction for microphones. Out-of-beam directions get attenuation. It's feasible with notebooks if the direction of interest is towards camera look direction. Currently that needs a custom topology for specific microphones distance and microphones count. If you are willing to experiment with non-mature component I can help to set it up. There will be later in future a noise suppressor for voice.

If you get sound from microphone capture it's worth to check that every volume/gain is sane. E.g. no gains competing against each other. If can report which topology is loaded (or attach dmesg) and which device you are capturing would help.

**Comment by marc-hb** [comment]
> @joakim-tjernlund please file an issue in the Linux tree

Transferring to Linux tracker
